rustls/src/msgs/deframer.rs

```diff
@@ -440,6 +399,54 @@ impl DeframerVecBuffer {
     }
 }
 
+trait DeframerBuffer {
+    fn advance(&mut self, num_bytes: usize);
```

Comment:
I think we should keep empty lines between these even if we're just declaring the interfaces. I think `copy()` should also probably get a docstring.

(Please maintain order as previously explained.)